openhands/cli/commands.py

```diff
@@ -327,3 +409,432 @@ def check_folder_security_agreement(config: OpenHandsConfig, current_dir: str) -
+    except Exception as e:
+        print_formatted_text(f'Error adding MCP server: {e}')
+
+
+async def add_sse_server(config: OpenHandsConfig) -> None:
+    """Add an SSE MCP server."""
+    print_formatted_text('Adding SSE MCP Server')
+
+    while True:  # Retry loop for the entire form
+        # Collect all inputs
+        url = await collect_input(config, '\nEnter server URL:')
+        if url is None:
+            print_formatted_text('Operation cancelled.')
+            return
+
+        api_key = await collect_input(
+            config, '\nEnter API key (optional, press Enter to skip):'
+        )
+        if api_key is None:
+            print_formatted_text('Operation cancelled.')
+            return
+
+        # Convert empty string to None for optional field
+        api_key = api_key if api_key else None
+
+        # Validate all inputs at once
+        try:
+            server = MCPSSEServerConfig(url=url, api_key=api_key)
+            break  # Success - exit retry loop
+
+        except ValidationError as e:
+            # Show all errors at once
+            print_formatted_text('❌ Please fix the following errors:')
+            for error in e.errors():
+                field = error['loc'][0] if error['loc'] else 'unknown'
+                print_formatted_text(f'  • {field}: {error["msg"]}')
+
+            if cli_confirm(config, '\nTry again?') != 0:
+                print_formatted_text('Operation cancelled.')
+                return
+
+    # Save to config file
+    config_data = load_config_file()
+    if 'mcp' not in config_data:
+        config_data['mcp'] = {}
+    if 'sse_servers' not in config_data['mcp']:
+        config_data['mcp']['sse_servers'] = []
+
+    server_config = {'url': server.url}
+    if server.api_key:
+        server_config['api_key'] = server.api_key
+
+    config_data['mcp']['sse_servers'].append(server_config)
+    save_config_file(config_data)
+
+    print_formatted_text(
+        f'✓ SSE MCP server added to ~/.openhands/config.toml: {server.url}'
+    )
+
+    # Prompt for restart
+    if await prompt_for_restart(config):
+        restart_cli()
+
+
+async def add_stdio_server(config: OpenHandsConfig) -> None:
+    """Add a Stdio MCP server."""
+    print_formatted_text('Adding Stdio MCP Server')
+
+    # Get existing server names to check for duplicates
+    existing_names = [server.name for server in config.mcp.stdio_servers]
+
+    while True:  # Retry loop for the entire form
+        # Collect all inputs
+        name = await collect_input(config, '\nEnter server name:')
+        if name is None:
+            print_formatted_text('Operation cancelled.')
+            return
+
+        command = await collect_input(config, "\nEnter command (e.g., 'uvx', 'npx'):")
+        if command is None:
+            print_formatted_text('Operation cancelled.')
+            return
+
+        args_input = await collect_input(
+            config, '\nEnter arguments (comma-separated, optional):'
```

Comment:
Comma-separated is pretty not convenient, can we accept `-y mcp-remote https://mcp.tavily.com/mcp/?tavilyApiKey=tvly-dev-XXX` and we do the split on our end?

After putting these in (no comma separated), i got these error:

<img width="2552" height="1266" alt="Image" src="https://github.com/user-attachments/assets/06bb2087-bd1b-4391-b2d6-3430dd507ae8" />